packages/firestore/lib/index.d.ts

```diff
@@ -430,14 +486,26 @@ export namespace Firestore {
     update(field: string | FieldPath, value: any, ...moreFieldsAndValues: any[]): Promise<void>;
   }
 
+  namespace DocumentReference {
+    type ObserverOnNext = (documentSnapshot: DocumentSnapshot) => void;
+    type ObserverOnError = (err: Query.SnapshotError) => void;
+    type ObserverOnComplete = () => void;
+
+    interface Observer {
+      next: ObserverOnNext;
+      error?: ObserverOnError;
+      complete?: ObserverOnComplete;
+    }
+  }
+
   /**
    * A DocumentSnapshot contains data read from a document in your Firestore database. The data can be extracted with
    * .`data()` or `.get(<field>)` to get a specific field.
    *
    * For a DocumentSnapshot that points to a non-existing document, any data access will return 'undefined'.
    * You can use the `exists` property to explicitly verify a document's existence.
    */
-  export interface DocumentSnapshot {
+  interface BaseDocumentSnapshot {
```

Comment:
Should this be exported?

Reply:
I couldn't think of any reason why the consumer would want to interact with it directly. It feels like an implementation detail, best left private if we want to change it later.

Reply:
The only thing is the docs would mention methods are extended from it - but it should handle that, just the user won't be able to see the interface.